cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java

```diff
@@ -270,8 +271,9 @@ public void initialize(MapReduceContext context) throws Exception {
 
   @Override
   public void destroy() {
-    finisher.onFinish(getContext().isSuccessful());
-    LOG.info("Batch Run finished : succeeded = {}", getContext().isSuccessful());
+    boolean isSuccessful = getContext().getState().getStatus() == ProgramStatus.COMPLETED;
+    finisher.onFinish(isSuccessful);
+    LOG.info("Batch Run finished : status = {}", getContext().getState());
```

Comment:
In `ETLMapReduce.destroy` method now, `isSuccessful` is computed based on the condition `getContext().getState().getStatus() == ProgramStatus.COMPLETED`